lib/Migration/Version18000Date20230504205823.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2023, Joas Schilling <coding@schilljs.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Talk\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\Types;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version18000Date20230504205823 extends SimpleMigrationStep {
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure(): ISchemaWrapper $schemaClosure
+	 * @param array $options
+	 * @return ?ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		if (!$schema->hasTable('talk_bots_server')) {
+			$table = $schema->createTable('talk_bots_server');
+			$table->addColumn('id', Types::BIGINT, [
+				'autoincrement' => true,
+				'notnull' => true,
+				'length' => 20,
+			]);
+			$table->addColumn('name', Types::STRING, [
+				'length' => 64,
+			]);
+			$table->addColumn('url', Types::STRING, [
+				'length' => 4000,
+			]);
+			$table->addColumn('url_hash', Types::STRING, [
+				'length' => 64,
+			]);
+			$table->addColumn('description', Types::STRING, [
+				'length' => 4000,
+				'notnull' => false,
+			]);
+			$table->addColumn('secret', Types::STRING, [
+				'length' => 128,
+			]);
+			$table->addColumn('error_count', Types::BIGINT, [
```

Comment:
Can we really have so many errors? ^^
I guess we can use `INT`, maybe with `'unsigned' => true` (even if it's not supported by all vendors)

`'unsigned' => true` could also be added to other integer fields

Reply:
basically I'm just unwilling to risk it. if a bot for whatever reason errors and is set on a million user instance for every room, it could just grow too quickly